test/DGmethods/sphere/advection_sphere_ssp34.jl

```diff
@@ -0,0 +1,385 @@
+# Advection on the Sphere test case.
+#
+# This is a case similar to the one presented in:
+# @article{WILLIAMSON1992211,
+#   title = {A standard test set for numerical approximations to the
+#            shallow water equations in spherical geometry},
+#   journal = {Journal of Computational Physics},
+#   volume = {102},
+#   number = {1},
+#   pages = {211 - 224},
+#   year = {1992},
+#   doi = {10.1016/S0021-9991(05)80016-6},
+#   url = {https://doi.org/10.1016/S0021-9991(05)80016-6},
+# }
+#
+# This version runs the advection on the sphere as a stand alone test (no
+# dependence on CLIMA moist thermodynamics)
+#--------------------------------#
+#--------------------------------#
+# Can be run with:
+# Integration Testing: JULIA_CLIMA_INTEGRATION_TESTING=true mpirun -n 1 julia --project=@. advection_sphere_ssp34.jl
+# No Integration Testing: JULIA_CLIMA_INTEGRATION_TESTING=false mpirun -n 2 julia --project=@. advection_sphere_ssp34.jl
+#--------------------------------#
+#--------------------------------#
+# Can be run on the GPU with:
+# Integration Testing: JULIA_CLIMA_INTEGRATION_TESTING=true mpirun -n 1 julia --project=/home/fxgiraldo/CLIMA/env/gpu advection_sphere_ssp34.jl
+# No Integration Testing: JULIA_CLIMA_INTEGRATION_TESTING=false mpirun -n 1 julia --project=/home/fxgiraldo/CLIMA/env/gpu advection_sphere_ssp34.jl
+#--------------------------------#
+#--------------------------------#
+
+using MPI
+using CLIMA.Topologies
+using CLIMA.Grids
+using CLIMA.DGBalanceLawDiscretizations
+using CLIMA.DGBalanceLawDiscretizations.NumericalFluxes
+using CLIMA.MPIStateArrays
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.StrongStabilityPreservingRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+using CLIMA.Vtk
+using LinearAlgebra
+using StaticArrays
+using Logging, Printf, Dates
+using Random
+
+@static if Base.find_package("CuArrays") !== nothing
```

Comment:
if #246 lands first this should be
```suggestion
@static if haspkg("CUDAnative")
 ```